File: LeetCode/1328-break-a-palindrome.py
###############################################################################################
# 简单，就三种情况，处理掉就行
###########
# 时间复杂度：O(n)
# 空间复杂度：O(1)
###############################################################################################
class Solution:
    def breakPalindrome(self, palindrome: str) -> str:
        if len(palindrome) == 1:
            return ""
        palindrome = list(palindrome)
        for i in range(len(palindrome)):
            if palindrome[i] != 'a' and ((not len(palindrome)%2) or i != len(palindrome)//2):
                palindrome[i] = 'a'
                return ''.join(palindrome)
        palindrome[-1] = 'b'
        return ''.join(palindrome)

File: LeetCode/1328-break-a-palindrome.cpp
///////////////////////////////////////////////////////////////////////////////////////////////
// easy
////////////////
// Time Complexity: O(n)
// Space Complexity: O(1)
///////////////////////////////////////////////////////////////////////////////////////////////
class Solution {
public:
    string breakPalindrome(string palindrome) {
        int n = palindrome.size();
        // no way to change it
        if (n < 2) return "";
        
        // change the first "not a" to a
        for (int i = 0; i < n / 2; ++i) {
            if (palindrome[i] != 'a') {
                palindrome[i] = 'a';
                return palindrome;
            }
        }
        // if all "a", change last one to "b"
        palindrome[n - 1] = 'b';
        return palindrome;
    }
};
